### PR TITLE
Add notice about migration versioning in 5.1

### PR DESCRIPTION
**Pull request by piotrj**
### Summary
When I started using 5.1 it was very surprising for me that I got an
error when I tried to run my migrations. And there is not that many
information about what happened. There is some blog post somewhere but
nothing on the official channel. I guess it would be useful to have it
here.

### CODE REVIEW

**Comment by rafaelfranca** [comment]
> When I started using 5.1 it was very surprising for me that I got an
> error when I tried to run my migrations.

Was not the the deprecation being show?

**Comment by rafaelfranca** [comment]
I don't think this is necessary. In Rails 5 when you inherit from `Migration` without a version you would see a deprecation. That deprecation has instructions how to fix it. That should be fixed before even trying to upgrade to 5.1 so I don't think the upgrade guide should mention something that is deprecated and was removed.

@matthewd thoughts?

**Comment by piotrj** [comment]
@rafaelfranca tested on all 5.0.x versions and didn't get deprecation notice on any of them when trying to migrate migration that was just ActiveRecord::Migration.

I actually bumped into this in gem that I was building, so it's the regular way someone would notice that. But anyway, there is no info anywhere about this.

**Comment by rafaelfranca** [comment]
I see. The deprecation is there but it is going to the logs:

```
$ bin/rails db:migrate
== 20170602181917 CreateUsers: migrating ======================================
-- create_table(:users, {})
   -> 0.0010s
== 20170602181917 CreateUsers: migrated (0.0011s) =============================

[2.4.0 foo (master #)]$ cat log/development.log
   (1.3ms)  CREATE TABLE "schema_migrations" ("version" varchar NOT NULL PRIMARY KEY)
   (1.1ms)  CREATE TABLE "ar_internal_metadata" ("key" varchar NOT NULL PRIMARY KEY, "value" varchar, "created_at" datetime NOT NULL, "updated_at" datetime NOT NULL)
  ActiveRecord::SchemaMigration Load (0.1ms)  SELECT "schema_migrations".* FROM "schema_migrations"
Migrating to CreateUsers (20170602181917)
   (0.1ms)  begin transaction
DEPRECATION WARNING: Directly inheriting from ActiveRecord::Migration is deprecated. Please specify the Rails release the migration was written for:

  class CreateUsers < ActiveRecord::Migration[4.2] (called from require at bin/rails:4)
   (0.4ms)  CREATE TABLE "users" ("id" INTEGER PRIMARY KEY AUTOINCREMENT NOT NULL, "name" varchar, "created_at" datetime, "updated_at" datetime)
  SQL (0.1ms)  INSERT INTO "schema_migrations" ("version") VALUES (?)  [["version", "20170602181917"]]
   (0.8ms)  commit transaction
  ActiveRecord::InternalMetadata Load (0.1ms)  SELECT  "ar_internal_metadata".* FROM "ar_internal_metadata" WHERE "ar_internal_metadata"."key" = ? LIMIT ?  [["key", "environment"], ["LIMIT", 1]]
   (0.1ms)  begin transaction
  SQL (0.3ms)  INSERT INTO "ar_internal_metadata" ("key", "value", "created_at", "updated_at") VALUES (?, ?, ?, ?)  [["key", "environment"], ["value", "development"], ["created_at", "2017-06-02 18:19:42.496114"], ["updated_at", "2017-06-02 18:19:42.496114"]]
   (0.7ms)  commit transaction
  ActiveRecord::SchemaMigration Load (0.1ms)  SELECT "schema_migrations".* FROM "schema_migrations"
```

I guess that is what we need to fix.

**Comment by kamipo** [comment]
#25297 will fix the issue.

**Comment by piotrj** [comment]
Ok, so let me know. If everyone thinks that it is not needed then we can close it. Nevertheless, more information is always better than less :)

File: guides/source/upgrading_ruby_on_rails.md [hunk 1]
```diff
@@ -100,6 +100,21 @@ To:
 Rails.application.secrets[:smtp_settings][:address]
 ```
 
+### Since 5.1 migrations have to be versioned
+
+In 5.0 when you generated migration it would get a version (the migration class
```
Comment (gmcgibbon) [inline guides/source/upgrading_ruby_on_rails.md]: If I could make a copy suggestion, I would rewrite this as

As of Rails 5.0, generated migrations inherit from a specific version of ActiveRecord's migration class (eg. `ActiveRecord::Migration[5.0]`). Versioning was introduced as a means to keep migration SQL consistent across Rails upgrades. Migrations generated using Rails 4 and below will need to be updated to inherit from a versioned migration class (eg. `ActiveRecord::Migration[4.2]`) in order to runnable on Rails 5.1.

**Review by gmcgibbon — COMMENTED** [review]
`5.0.7` appears to still be logging this deprecation message. I don't see any mention of versioned migrations in the current upgrade docs. Since it was decided to keep using `ActiveSupport::Deprecation.warn`, it makes sense to proceed with this PR, right?

**Pull request closed without merging**